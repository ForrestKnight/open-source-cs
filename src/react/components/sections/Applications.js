/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Applications extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Applications" className="Applications">
                
            </div>
        );
    }
}


export default Applications;