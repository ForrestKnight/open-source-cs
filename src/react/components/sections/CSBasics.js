/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class CSBasics extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="CSBasics" className="CSBasics">
                
            </div>
        );
    }
}


export default CSBasics;