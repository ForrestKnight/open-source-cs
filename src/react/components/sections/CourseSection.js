/* Packages */
import React from 'react';
import Course from '../Course';


/* Component */
export class CourseSection extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id={this.props.name} className="CourseSection">
                
            </div>
        );
    }
}


export default Math;