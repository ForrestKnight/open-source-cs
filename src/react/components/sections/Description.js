/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Description extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Description" className="Description">
                
            </div>
        );
    }
}


export default Description;