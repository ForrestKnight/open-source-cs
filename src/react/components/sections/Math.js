/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Math extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Math" className="Math">
                
            </div>
        );
    }
}


export default Math;