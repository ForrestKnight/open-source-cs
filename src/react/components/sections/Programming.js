/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Programming extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Programming" className="Programming">
                
            </div>
        );
    }
}


export default Programming;