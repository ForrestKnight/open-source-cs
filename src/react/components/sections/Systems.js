/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Systems extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Systems" className="Systems">
                
            </div>
        );
    }
}


export default Systems;