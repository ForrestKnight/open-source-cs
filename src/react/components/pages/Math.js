/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import MathCourses from '../../data/MathCourses';


/* Component */
export class Math extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Math');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Math" className="page">
                <CourseSection 
                    name="Math"
                    courses={MathCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Math);