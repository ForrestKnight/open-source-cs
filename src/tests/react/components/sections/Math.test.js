/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {Math} from '../../../../react/components/sections/Math';


/* Test Config */
let component;
beforeEach(() => {
    component = shallow(<Math />);
});


/* Tests */
test('Render Math section', () => {
    expect(component).toMatchSnapshot();
});

