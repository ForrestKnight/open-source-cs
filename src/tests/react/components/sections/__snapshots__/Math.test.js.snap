// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Math section 1`] = `
<div
  className="Math"
  id="Math"
/>
`;
